draft-jholland-quic-multicast.md

```diff
@@ -209,6 +209,11 @@ Note that the hash is on the unencrypted packet because it checks against a spec
 
 # Packet Scheduling
 
+# Stateless Reset
+
+As clients can unilaterally stop the delivery of multicast packets by leaving the relevant Group, channels do not need stateless reset tokens.
+Instead, if an endpoint receives packets addressed to a group IP that it can not associate with any existing channel, it MAY take the necessary steps to prevent the reception of further such packets. For example, it might issue an IGMP or MLD report indicating a desire to leave the (S,G) associated with the unwanted packet.
```

Comment:
Almost there, but I'll suggest:

OLD:
~~~
As clients can unilaterally stop the delivery of multicast packets by leaving the
relevant Group, channels do not need stateless reset tokens.
Instead, if an endpoint receives packets addressed to a group IP that it can not
associate with any existing channel, it MAY take the necessary steps to prevent
the reception of further such packets. For example, it might issue an IGMP or
MLD report indicating a desire to leave the (S,G) associated with the unwanted
packet.
~~~

NEW:
~~~
As clients can unilaterally stop the delivery of multicast packets by leaving
channels associated with network paths that receive traffic with unknown
Channel IDs, channels do not need stateless reset tokens.
Instead, if an endpoint receives packets addressed to a group IP that it can
not associate with any existing channel, it MAY leave channels using that
group IP with reason "spurious traffic".
~~~

I think it's roughly equivalent in the end result, but I'm trying to talk in terms of the quic channel associated with the network path instead of the IP multicast group to the extent possible.

Also maybe as a separate but related matter we should clarify somewhere that the client MUST (or just SHOULD?) notify the server with a MC_CLIENT_CHANNEL_STATE frame when it joins and leaves channels associated with addresses of defined quic channels?  (The OLD text seems to indicate just stopping the receive at the IGMP/UDP layer without sending the MC_CLIENT_CHANNEL_STATE frame would be reasonable, and while that will eventually hit the max_idle, I don't think it's a good practice and shouldn't be encouraged--they should do a proper quic leave of the channel IMO.)

Also worth noting that I'm using plural here on purpose--maybe this points to another thing that (as another separate but loosely related matter) should get some text on how to handle it when there's colliding network paths among the channels?  If that makes any sense I guess we can make an issue about it...
 - Note that there's actually a use case for permitting (S,G) collisions: it's useful to deliver integrity frames for your data channel on a different quic channel that uses the same (S,G) (with or without being a different UDP port), because it uses the same routing for both channels so there's fate-sharing for the integrity and data trafic, and that can give a bunch of benefit to the scalability by shipping the integrity traffic over multicast.